@here/olp-sdk-dataservice-read/lib/utils/getTile.ts

```diff
@@ -61,109 +118,202 @@ import {
  *
  * ```
  *
- * @returns The blob of the tile.
+ * @returns The blob of the requested Tile or the blob of the closest parent Tile.
  */
 export async function getTile(
-    rq: TileRequest,
+    request: TileRequest,
     abortSignal?: AbortSignal
+): Promise<Response>;
+
+export async function getTile(
+    request: TileRequest,
+    paramsOrSignal?: TileRequestParams | AbortSignal,
+    signal?: AbortSignal
 ): Promise<Response> {
-    const layerType = rq.getParams().layerType;
-    const settings = rq.getParams().settings;
-    const catalogHrn = rq.getParams().catalogHrn;
-    const layerId = rq.getParams().layerId;
-    const billingTag = rq.getBillingTag();
-    const quadKey = rq.getTileKey();
-    const fetchOptions = rq.getFetchOption();
-    const cache = new MetadataCacheRepository(settings.cache);
+    let params: TileRequestParams | undefined;
+    let abortSignal: AbortSignal | undefined;
+
+    if (
+        !paramsOrSignal ||
+        (paramsOrSignal && paramsOrSignal instanceof AbortSignal)
+    ) {
+        // usind deprecated
+        params = request.getParams();
+        abortSignal = paramsOrSignal;
+    } else {
+        params = paramsOrSignal;
+        abortSignal = signal;
+    }
 
+    const quadKey = request.getTileKey();
     if (!quadKey) {
         return Promise.reject(new Error("Please provide correct QuadKey"));
     }
 
-    const partitionId = `${mortonCodeFromQuadKey(quadKey)}`;
-    const partitionsRequest = new PartitionsRequest().withPartitionIds([
-        partitionId
-    ]);
+    if (!params) {
+        return Promise.reject(
+            new Error(
+                `Error getting params. Use TileRequest.withParams() method.`
+            )
+        );
+    }
+
+    let catalogVersion: number | undefined;
+    let blobType: ApiName = "volatile-blob";
+
+    if (params.layerType === "versioned") {
+        blobType = "blob";
+        catalogVersion = await request.getCatalogVersion();
+    }
 
-    const blobType = layerType === "versioned" ? "blob" : "volatile-blob";
-    const queryClient = new QueryClient(settings);
     const blobRequestBuilder = await RequestFactory.create(
         blobType,
         "v1",
-        settings,
-        catalogHrn,
+        params.settings,
+        params.catalogHrn,
         abortSignal
     ).catch(error => Promise.reject(error));
 
-    if (fetchOptions !== FetchOptions.OnlineOnly) {
+    const delta = 4;
+    const requestedTileKey = TileKey.fromRowColumnLevel(
+        quadKey.row,
+        quadKey.column,
+        quadKey.level
+    );
+    const parentTileKey = requestedTileKey.changedLevelBy(-delta);
+    const subQuadKey = requestedTileKey.getSubHereTile(delta);
+
+    if (request.getFetchOption() !== FetchOptions.OnlineOnly) {
+        const cache = new MetadataCacheRepository(params.settings.cache);
+        const partitionId = subQuadKey;
```

Comment:
I am not sure we should use subQuadKey for storing the exploded partitions in cache as this is always relative to the root, but you root might be different. E.g. you request first on level 14, root is on 10, then you request something on level 13, root is on 9, but actually you stored level 13 partitions with root on level 10 as reference. Means you will not find anything as long as you are not on the same level. Better is to save the full tilekey in cache instead of the subQuadKey.

Reply:
I see. Thanks!

Reply:
updated